src/datetimepicker.android.js

```diff
@@ -109,7 +110,15 @@ export default function RNDateTimePicker(props: AndroidNativeProps) {
 
           switch (action) {
             case DATE_SET_ACTION:
-              event.nativeEvent.timestamp = date.setFullYear(year, month, day);
+              date.setFullYear(year, month, day);
+              date.setHours(0, 0, 0);
+              if (timeZoneOffsetInMinutes !== undefined) {
```

Comment:
What do you think to move the offset caculation to another function to make more easier to test and reuse the code